packages/zk-certificates/contracts/AirdropGateway.sol

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "./interfaces/IZkKYCVerifier.sol";
+
+/**
+ * @title AirdropGateway: manages airdrops for the Galactica
+ */
+contract AirdropGateway is AccessControl {
+    // roles for access control
+    bytes32 public constant CLIENT_ROLE = keccak256("CLIENT_ROLE");
+    uint public distributionIndexCounter;
+    // due to some Solidity peculiarity, the associated getter doesn't return the dynamic array of requiredSBTs
+    mapping(uint => AirdropDistribution) public distributions;
+    // distributionIndex -> userAddress -> registration status
+    mapping(uint => mapping(address => bool)) public registeredUsers;
+    // distribution Index -> humanId => registration status
+    mapping(uint => mapping(bytes32 => bool)) public registeredHumanID;
+    // distributionIndex -> userAddress -> claimed status
+    mapping(uint => mapping(address => bool)) public claimedUsers;
+    IZkKYCVerifier public verifierWrapper;
```

Comment:
This works with one global verifier for every distribution. I am wondering though
1. If some distributions might use different verifiers. Maybe there will be one distribution for non-US residents and one for SwissBorg KYC holders.
2. If it would make sense to reuse a zkKYC the user already has completed by checking the VerificationSBT instead of requiring a new proof. This could be a bit simpler for users that already completed the Galactica onboarding.

I guess these are more questions about the product design instead of the implementation @mike-tis

Reply:
Nick told me that we need to differentiate users based on humanID and to do that I need to check user zkKYC proof, this verifier here refers to that.